psi4/src/psi4/detci/ints.cc

```diff
@@ -131,66 +133,69 @@ void CIWavefunction::setup_dfmcscf_ints() {
     outfile->Printf("\n   ==> Setting up DF-MCSCF integrals <==\n\n");
 
     /// Build JK object
-    if (options_.get_str("SCF_TYPE") == "DF"){
+    if (options_.get_str("SCF_TYPE") == "DF") {
         jk_ = JK::build_JK(basisset_, get_basisset("DF_BASIS_SCF"), options_);
     } else {
         jk_ = JK::build_JK(basisset_, BasisSet::zero_ao_basis_set(), options_);
     }
+
     jk_->set_do_J(true);
     jk_->set_do_K(true);
-    jk_->initialize();
     jk_->set_memory(Process::environment.get_memory() * 0.8 / sizeof(double));
+    jk_->initialize();
     jk_->print_header();
 
+    int num_threads_ = 1;
+#ifdef _OPENMP
+    num_threads_ = omp_get_max_threads();
+#endif
+
     /// Build DF object
-    dferi_ = DFERI::build(get_basisset("ORBITAL"), get_basisset("DF_BASIS_SCF"), options_);
-    dferi_->set_memory(Process::environment.get_memory() * 0.8 / sizeof(double));
-    dferi_->print_header();
+    // ==> Init DF object <== /
+    dfh_ = std::shared_ptr<df_helper::DF_Helper>(
+        new df_helper::DF_Helper(get_basisset("ORBITAL"), get_basisset("DF_BASIS_SCF")));
+    dfh_->set_memory(Process::environment.get_memory() * 0.8 / sizeof(double));
```

Comment:
As above, this could be cleaned up by using `make_shared` to instantiate `dfh_`.